### ISSUE TITLE
Automatically resolve spelt out Emoji names to Emoji characters

### ISSUE THREAD

**Issue by Empty2k12** [vector-im/riot-meta#211]
This is an issue that has been bugging me on all Riot platforms:

Usually in software where you can use emoji, they are automatically resolved from the spelt out name: (e.g `:joy:` to 😂, `:+1:` to 👍, ...). This works across all platforms and various programs such as Slack, GitHub, WhatsApp, Hipchat, ...

In Riot, typing `:joy:` does not resolve to the corresponding Emoji.

There are Emoji name mappings available here: https://www.webpagefx.com/tools/emoji-cheat-sheet/

**Comment by aaronraimist** [comment]
This does work for you in Riot Web, correct? This is only for the mobile clients.

**Comment by Empty2k12** [comment]
Nope. Writing `:thumbsup:` does not convert to the Emoji. You need to insert the literal Emoji into the text for it to show.

**Comment by aaronraimist** [comment]
![screen shot 2018-12-17 at 11 51 13 am](https://user-images.githubusercontent.com/5855073/50105399-34240500-01f2-11e9-82d3-dc17b1632f72.png)

and then you press tab to select the one you want. You don't see that? Are you running version 0.17.8?

**Comment by Empty2k12** [comment]
I am seeing that, but that's not the bahviour I am requesting.

In Slack and other applications typing out `:thumbsup:` and sending that message, the literal string `:thumbsup:` will convert to the 👍 Emoji. Copying the text containing the Emoji, the text placed in the clipboard will have the emoji replaced by the literal text `:thumbsup:` again. 

You're suggesting I use the Auticompletion to replace the part of :emojinam... that I typed so far.

**Comment by aaronraimist** [comment]
Presumably that is what the "Automatically replace plain text Emoji" option is supposed to do but it seems to be broken https://github.com/vector-im/riot-web/issues/4872

**Comment by Empty2k12** [comment]
Correct, this is the exact thing.

However, this would need to be implemented for all platforms, not just Web.
